src/plugins/blocksmith/components/modal-error-boundary/index.jsx

```diff
@@ -0,0 +1,47 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import swal from 'sweetalert2';
+
+class ModalErrorBoundary extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    onCloseModal: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    const { onCloseModal } = this.props;
+    console.error(error, errorInfo);
```

Comment:
@efjacobson - is this intentional?

Reply:
@arunkarnati i suppose its not necessary, it will get tied into the api error reporting eventually. removed for now